src/main/kotlin/org/rust/ide/intentions/createFromUsage/CreateFunctionIntention.kt

```diff
@@ -103,12 +100,23 @@ class CreateFunctionIntention : RsElementBaseIntentionAction<CreateFunctionInten
             val target = getTargetItemForFunctionCall(path) ?: return null
             val name = path.referenceName ?: return null
 
-            return if (target is CallableInsertionTarget.Item) {
-                text = "Create associated function `${target.item.name}::$name`"
-                Context.AssociatedFunction(functionCall, name, target.module, target.item)
-            } else {
-                text = "Create function `$name`"
-                Context.Function(functionCall, name, target.module)
+            return when (target) {
+                is RsMod -> {
+                    text = "Create function `$name`"
+                    Context.Function(functionCall, name, target)
+                }
+
+                is RsStructOrEnumItemElement -> {
+                    text = "Create associated function `${target.name}::$name`"
+                    Context.AssociatedFunction(functionCall, name, target.containingMod, target, existingImpl = null)
```

Comment:
It looks like in the current usage, `existingImpl` and `item` are exclusive? It could be modeled by some enum, but it's probably overkill.